pkg/api/v1alpha1/cluster.go

```diff
@@ -309,6 +310,17 @@ func (c *Cluster) ClearPauseAnnotation() {
 	}
 }
 
+// ExperimentalSelfManagedClusterUpgrade returns true if
+// ExperimentalSelfManagedClusterUpgrade is active, cluster is self managed
+// and the provider is vSphere
+// in future we will remove this method once we have kindless upgrade for all the providers.
+func (c *Cluster) ExperimentalSelfManagedClusterUpgrade() bool {
```

Comment:
can we have this method implemented somewhere else? i don't think it belongs here, and even less as cluster method